### ISSUE TITLE
No valid molecules being generated

### ISSUE THREAD

**Issue by Arpit-Avasarmol-git** [ComputeSuda/DIFFUMOL#3]
I have trained the model with following configs:
{"lr":0.0001,"batch_size":2048,"microbatch":64,"learning_steps":100000,"log_interval":2000,"save_interval":10000,"eval_interval":10000,"ema_rate":"0.9999","resume_checkpoint":"none","schedule_sampler":"lossaware","diffusion_steps":2000,"noise_schedule":"sqrt","timestep_respacing":"","vocab":"./datasets/vob.txt","use_plm_init":"no","vocab_size":0,"config_name":"./datasets/model.json","notes":"folder-notes","data_path":"./datasets/moses2.csv","data_name":"moses2","num_props":2,"props":["qed","logp"],"scaffold":false,"checkpoint_path":"./weight_conditional_property","seq_len":128,"hidden_t_dim":128,"hidden_dim":128,"dropout":0.1,"use_fp16":false,"fp16_scale_growth":0.001,"seed":102,"gradient_clipping":-1.0,"weight_decay":0.0,"learn_sigma":false,"use_kl":false,"predict_xstart":true,"rescale_timesteps":true,"rescale_learned_sigmas":false,"sigma_small":false,"emb_scale_factor":1.0}

Once, after training when I'm going for generating there no valid molecules being generated. Can you pls tell me how do I generate valid molecules or anything I can do so that valid molecules starts to appear.

I'm getting the following logs:
### tokenized_datasets...example [0.4, 3.0, 2.0]
RAM used: 1671.58 MB
merge and mask (num_proc=1): 100%|████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [00:00<00:00, 218.13 examples/s]
RAM used: 1671.58 MB
padding (num_proc=1): 100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [00:00<00:00, 217.79 examples/s]
Dataset({
    features: ['input_id_x', 'input_id_y', 'input_ids', 'input_mask'],
    num_rows: 100
}) padded dataset
RAM used: 1671.58 MB
RAM used: 1671.58 MB
### End of reading iteration...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [02:28<00:00, 148.74s/it]
### Written the decoded output to /ganga/narmada/Diffumol-1/DIFFUMOL/output/weight_conditional_property
nums of smiles: 100
nums of molecules: 0

### First 10 generated SMILES:
1. C = C C C C N ( C ) C ( = O ) c 1 c c c 2 c c ( C ( = O ) O C ) c c c 2 n 1 [END]
2. C C ( = O ) c 1 c c ( - c 2 n n c ( C S c 3 n c c c n 3 ) o 2 ) o c 1 C [END]
3. C C ( = O ) c 1 c c c ( O C ( = O ) C N 2 C ( C ) C C C 2 C ) c ( C ) c 1 [END]
4. C O C ( = O ) C C C C S c 1 n c 2 c c c c c 2 o 1 [END]
5. C C C O C ( = O ) C C C ( = O ) c 1 c c ( F ) c ( F ) c c 1 F [END]
6. O = C ( O c 1 c c c c 2 c c c c c 1 2 ) c 1 c n ( - c 2 c c n c c 2 ) n n 1 [END]
7. C C S c 1 n c ( O C ( C ) = O ) c ( N 2 C C C C 2 ) c ( C ) n 1 [END]
8. C C ( = O ) c 1 c c c c c 1 O C C ( = O ) c 1 c n c 2 c c c c c 2 n 1 [END]
9. C # C C O C ( = O ) c 1 c c c 2 o c ( C ) c c 2 c 1 [END]
10. C O c 1 c ( Cl ) c c c c 1 S C C S c 1 n c n [nH] 1 [END]
No valid molecules generated!
Total SMILES generated: 100
Valid molecules: 0
[rank0]:[W1117 12:55:05.181521859 ProcessGroupNCCL.cpp:1524] Warning: WARNING: destroy_process_group() was not called before program exit, which can leak resources. For more info, please see https://pytorch.org/docs/stable/distributed.html#shutdown (function operator())

**Comment by Arpit-Avasarmol-git** [comment]
We can just change the parsing method as rdkit takes the space as invalid molecules, change the regex function, and now it generates the valid molecules.

(diffumol-1) root@merai_aa57114v:/ganga/narmada/Diffumol-1/DIFFUMOL# python3 generate.py --model_path /ganga/narmada/Diffumol-1/DIFFUMOL/weight_conditional_property/ema_0.9999_050000.pt --batch_size 256 --sample 10 --out_dir /ganga/narmada/Diffumol-1/DIFFUMOL/output --value 0.9 --valuebck 1.5
Logging to /tmp/openai-2025-11-17-14-52-27-106051
0
### Generating molecules with properties:
### QED (drug-likeness): 0.9
### LogP (lipophilicity): 1.5
/ganga/narmada/Diffumol-1/DIFFUMOL/weight_conditional_property/training_args.json
### Creating model and diffusion...
### The parameter count is 87301220
############################## load vocab from ./datasets/vob.txt
### Sampling...on test
############################## 
Loading text data...
############################## 
Loading dataset moses2 from ./datasets/moses2.csv...
### Data samples...
 [[0.9, 1.5], [0.9, 1.5]] ['[PAD]', '[PAD]']
RAM used: 1665.89 MB
Dataset({
    features: ['src', 'trg'],
    num_rows: 10
})
RAM used: 1667.89 MB
Running tokenizer on dataset (num_proc=4): 100%|████████████████████████████████████████████| 10/10 [00:00<00:00, 15.80 examples/s]
### tokenized_datasets Dataset({
    features: ['input_id_x', 'input_id_y'],
    num_rows: 10
})
### tokenized_datasets...example [0.9, 1.5]
RAM used: 1669.89 MB
merge and mask (num_proc=1): 100%|██████████████████████████████████████████████████████████| 10/10 [00:00<00:00, 23.66 examples/s]
RAM used: 1669.89 MB
padding (num_proc=1): 100%|█████████████████████████████████████████████████████████████████| 10/10 [00:00<00:00, 23.83 examples/s]
Dataset({
    features: ['input_id_x', 'input_id_y', 'input_ids', 'input_mask'],
    num_rows: 10
}) padded dataset
RAM used: 1669.89 MB
RAM used: 1669.89 MB
### End of reading iteration...
100%|████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:45<00:00, 45.04s/it]
### Total takes 45.05s .....
### Written the decoded output to /ganga/narmada/Diffumol-1/DIFFUMOL/output/weight_conditional_property
nums of smiles: 10
nums of molecules: 10

### First 10 generated SMILES:
1. [START] C C ( C ) N S ( = O ) ( = O ) c 1 c c ( Cl ) c ( N 2 C C N C 2 = O ) s 1 [END]
2. [START] C C n 1 c c c ( S ( = O ) ( = O ) N c 2 c c c c 3 c 2 O C C O 3 ) n 1 [END]
3. [START] C O c 1 c c c ( N C ( = O ) C 2 C C C ( = O ) N ( C ) C 2 ) c ( O C ) c 1 [END]
4. [START] C O C C ( = O ) N 1 C C C ( O ) C C 1 C ( = O ) N c 1 c c c ( Cl ) c c 1 [END]
5. [START] C C ( C ) ( C O ) N C ( = O ) c 1 c c c 2 c c n ( C ) c 2 c 1 [END]
6. [START] C C ( C ) n 1 c n n c 1 C N C ( = O ) C N 1 C C C c 2 c c c c c 2 1 [END]
7. [START] C O C c 1 n c 2 c c ( C ( = O ) N C C 3 C C C O 3 ) c c c 2 [nH] 1 [END]
8. [START] C N c 1 c c ( C ( = O ) N 2 C C C ( C O ) C C 2 ) c 2 c c c n c 2 n 1 [END]
9. [START] C n 1 c ( C ( = O ) C N ) n c 2 c c ( C ( F ) ( F ) F ) c c c 2 1 [END]
10. [START] C O c 1 c c c ( C C N 2 C C O C C 2 ) c c 1 O C [END]

### First 10 processed SMILES (spaces removed):
1. CC(C)NS(=O)(=O)c1cc(Cl)c(N2CCNC2=O)s1
2. CCn1ccc(S(=O)(=O)Nc2cccc3c2OCCO3)n1
3. COc1ccc(NC(=O)C2CCC(=O)N(C)C2)c(OC)c1
4. COCC(=O)N1CCC(O)CC1C(=O)Nc1ccc(Cl)cc1
5. CC(C)(CO)NC(=O)c1ccc2ccn(C)c2c1
6. CC(C)n1cnnc1CNC(=O)CN1CCCc2ccccc21
7. COCc1nc2cc(C(=O)NCC3CCCO3)ccc2[nH]1
8. CNc1cc(C(=O)N2CCC(CO)CC2)c2cccnc2n1
9. Cn1c(C(=O)CN)nc2cc(C(F)(F)F)ccc21
10. COc1ccc(CCN2CCOCC2)cc1OC
[14:53:31] DEPRECATION WARNING: please use MorganGenerator
[14:53:31] DEPRECATION WARNING: please use MorganGenerator
[14:53:31] DEPRECATION WARNING: please use MorganGenerator
[14:53:31] DEPRECATION WARNING: please use MorganGenerator
[14:53:31] DEPRECATION WARNING: please use MorganGenerator
[14:53:31] DEPRECATION WARNING: please use MorganGenerator
[14:53:31] DEPRECATION WARNING: please use MorganGenerator
[14:53:31] DEPRECATION WARNING: please use MorganGenerator
[14:53:31] DEPRECATION WARNING: please use MorganGenerator
[14:53:31] DEPRECATION WARNING: please use MorganGenerator
Valid ratio:  1.0
Unique ratio:  1.0
Novelty ratio:  1.0
[rank0]:[W1117 14:53:32.381338968 ProcessGroupNCCL.cpp:1524] Warning: WARNING: destroy_process_group() was not called before program exit, which can leak resources. For more info, please see https://pytorch.org/docs/stable/distributed.html#shutdown (function operator())
(diffumol-1) root@merai_aa57114v:/ganga/narmada/Diffumol-1/DIFFUMOL#
